rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -178,18 +211,43 @@ public String loginWithCredential(@MultipartForm LoginForm multipart) throws Act
             NodeException, KeyException, IOException, LoginException, RMException {
 
         Session session;
+        CredData credData;
+
         if (multipart.getCredential() != null) {
-            session = sessionStore.createUnnamedSession();
             Credentials credentials = Credentials.getCredentials(multipart.getCredential());
+            String schedulerPrivateKey = PASchedulerProperties.getAbsolutePath(PASchedulerProperties.SCHEDULER_AUTH_PRIVKEY_PATH.getValueAsString());
+            credData = credentials.decrypt(schedulerPrivateKey);
```

Comment:
What about if credentials cannot be decrypted. Don't we need a try/catch or check here?

Reply:
If credentials cannot be decrypted, an exception will be thrown, and it is better to not catch it. All its details/trace will appear in the sent response.